blog/02-01-2023-flashloan-aave-celo/flashloan aave celo.md

```diff
@@ -0,0 +1,254 @@
+
+![remix metamask](remix.jpeg)
+
+## Step 1: Setting up your Code 
+
+To begin, you'll need to create a new contract that you will call and which will be called by the Aave Protocol's LendingPool contract. This can be two separate contracts, however, for ease of understanding, we'll consolidate those functions into one contract.
+
+In Remix, add a new file and name it, e.g. myFlashLoan.sol and add the following code to it:
+```
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import { FlashLoanReceiverBase } from "./FlashLoanReceiverBase.sol";
+import { ILendingPool, ILendingPoolAddressesProvider, IERC20 } from "./Interfaces.sol";
+import { SafeMath } from "./Libraries.sol";
+import "./Ownable.sol";
+/** 
+    !!!
+    Never keep funds permanently on your FlashLoanReceiverBase contract as they could be 
+    exposed to a 'griefing' attack, where an attacker uses the stored funds.
+    !!!
+ */
+contract MyV2FlashLoan is FlashLoanReceiverBase, Ownable {
+    using SafeMath for uint256;
+
+    constructor(ILendingPoolAddressesProvider _addressProvider) FlashLoanReceiverBase(_addressProvider) {}
+```
+
+Although remix can import solidity source files directly from GitHub, In the first 3 lines, we are locally importing the files necessary to interface with the Aave Protocol. You will want to create several files on that segue:
+
+* [FlashLoanReceiverBase.sol](https://gist.github.com/johnfawole/741aff645a7e375f8fbccb0fab5ee910)
+* [Interfaces.sol](https://gist.github.com/johnfawole/c2844bd1450607ead2969b8f83e340a3)
+* [Libraries.sol](https://gist.github.com/johnfawole/353940d436643123603a8375703646ec)
+* [Ownable.sol](https://gist.github.com/johnfawole/8e37c45d86315744339b952a2846ff84)
+
+![contracts](contracts.jpeg)
+
+The MyFlashLoan contract inherits from the FlashLoanReceiverBase and Ownable contract.
+
+On line 20, we are passing in a constructor argument of an address to the inherited FlashLoanReceiverBase contract which determines which LendingPoolAddressProvider to use on the network the contract is deployed. 
+
+We can find the _addressProvider from [the documentation](https://docs.aave.com/developers/v/2.0/deployed-contracts/deployed-contracts) as we'll be passing in the relevant LendingPoolAddressProvider value at deployment.
+
+## Step 2: Implementing your Flash Loan function
+Next, add the function that will be called to initiate the Flash Loan process. Let's call the function myFlashLoanCall() and this function will take the address of the token you want to flashloan as an argument. 
+
+We will limit access to it to the owner (i.e. you), this will help guard against griefing attacks. You will request a Flash Loan of 1 USDC (1* 10^6 as USDC has 6 decimal places). We don't need data for the flashloan, so we are passing an empty string. 
+
+This function then calls the flashloan() function on the lending pool contract for you to receive the tokens.
+```
```

Comment:
add a space before the code snippet and add the intended code language like this 
" ```solidity
// code here
``` "